### DISCUSSION TITLE
saving new fields in ply file

### DISCUSSION THREAD

**Discussion by ldv1** [pyvista/pyvista#3338] [Q&A] [answered]
hi,

I read in a ply file, compute a new field and save the result in a new ply file, but surprisingly the new field is not saved. Here is my piece of code:

```
import pyvista as pv

mesh = pv.read("bun_zipper.ply")
curv = mesh.curvature()
mesh.add_field_data(curv, 'mean_curvature')
mesh.save('bunny.ply', binary = False)
```

Thanks for the help!

**Comment by MatthewFlamm** [comment]
I'm not sure that `ply` supports this kind of data.  Is it required that you save the data in `ply` format?  If you save in a vtk native format like `vtp` or `vtk` the data should be saved.

**Reply by ldv1** [reply]
> I'm not sure that ply supports this kind of data

why not? in a ply fle, you can save any scalar at each node (= the signal); in most existing ply files, you will find e.g. the normals nx, ny, nz; there is nothing wrong with adding more fields. I am just surprised that my change to `mesh` isn't really considered, at least when I save it as a ply file. This is misleading.

My work-around is to code a small ply writer.

**Comment by MatthewFlamm** [comment, marked as answer]
The field data `'mean_curvature'` does not exist on the points.  To do this you would use `point_data`, e.g. `mesh.point_data["mean_curvature"]=curv`.

Sorry, I thought the data was a single value for the entire mesh, but it looks like it exists on the points.

It seems that the default behavior is to not write this data to the file when using `ply` format currently.  The vtk writer has some methods to control writing some data like this, but I don't think this is currently exposed in PyVista.  See https://vtk.org/doc/nightly/html/classvtkPLYWriter.html#a5c65f812d9c1bd918a6fd134236229dc as an example.

**Reply by ldv1** [reply]
Thank you for your investigation.
AFAIC, this issue is closed.

**Reply by MatthewFlamm** [reply]
Sorry that I responded 'out of thread' with my other comment, this is something I often miss.  I do think that PyVista could have a better interface for writing files.  This would go a long way towards addressing needs like this (if it is supported in vtk).  Maybe, something like:

```py
import pyvista as pv
mesh = pv.Sphere()
writer = mesh.get_writer('file.ply')
writer.set_some_option()
writer.write()
```
